api/candlepin-api-spec.yaml

```diff
@@ -30,6 +30,336 @@ paths:
+          in: path
+          description: The Id of activation key to be deleted
+          required: true
+          schema:
+            type: string
+      responses:
+        204:
+          description: A successful operation
+        400:
+          description: Activation key Id is null or empty or could not be found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Activation key Id is null or empty or could not be found
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+  /activation_keys/{activation_key_id}/pools:
+    get:
+      description: Retrieves a list of pools based on the Activation Key
+      tags:
+        - Activation Key
+      operationId: getActivationKeyPools
+      security: []
+      parameters:
+        - name: activation_key_id
+          in: path
+          description: The Id of activation key
+          required: true
+          schema:
+            type: string
+      x-java-response:
+        type: Iterable
+        isContainer: true
+      responses:
+        200:
+          description: List of pools based on the Activation Key
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/PoolDTO'
+        400:
+          description: Activation key Id is null or empty or could not be found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Activation key Id is null or empty or could not be found
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+  /activation_keys/{activation_key_id}/pools/{pool_id}:
+    post:
+      description: Adds a Pool to an Activation Key
+      tags:
+        - Activation Key
+      operationId: addPoolToKey
+      security: []
+      parameters:
+        - in: path
+          name: activation_key_id
+          description: Id of an activation key
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: pool_id
+          description: Id of pool
+          required: true
+          schema:
+            type: string
+        - in: query
+          name: quantity
+          schema:
+            type: integer
+            format: int64
+      responses:
+        200:
+          description: Pool added to activation key
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ActivationKeyDTO'
+        400:
+          description: Activation key Id is null or empty or either pool or activation key could not be
```

Comment:
This feels like another collection of all possible BadRequestExceptions. In this case it's better to be less explicit about any possible error (since that doesn't tell you anything anyway), and just focus on the type of issue it could be: invalid input or the pool could not be added.

Though, honestly, this is a case where maybe we consider changing the exception type and http code for the rules exception, as those really aren't bad requests; but that's a separate discussion that would involve Katello.

Reply:
Simplified the description.